File: algorithms_organized_in_pictures/chap05/InsertSort.py
def insertSort(x):
    for size in range(1, len(x)):
        val = x[size]
        i = size
    while i > 0 and x[i-1] > val:
        x[i] = x[i-1]
        i -= 1
    x[i] = val


x = [5, 2, 8, 6, 1, 9, 3, 7]
insertSort(x)
print(x)

File: _books/algorithms_organized_in_pictures/chap05/InsertSort.java
public class InsertSort {
    public static void main (String[] args) {
        int[] a = {12, 326, 127, 467, 110, 58};

        for (int j = 1; j < a.length; j++) {
            int key = a[j];
            int i = j - 1;
            while(i >= 0 && a[i] > key) {
                a[i + 1] = a[i];
                i = i - 1;
            }
            a[i + 1] = key;
        }

        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i] + " ");
        }
    }
}
